### PR TITLE
Upgrade 1.24 to kernel-5.10

### LINKED ISSUE #857: Upgrade Linux Kernel to 5.10
**What would you like to be added**:
Upgrade the Linux kernel to 5.10

**Why is this needed**:
[Amazon Linux 2 AMI has been upgraded to Linux 5.10](https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-linux-2-ami-kernel-5-10/), so I believe the EKS AMI should also be upgraded. This is also necessary as some features of Calico and Cilium, like WireGuard for encrypting traffic, require a newer version of the Linux kernel, as mentioned in [this](https://github.com/awslabs/amazon-eks-ami/issues/727) issue.

### PR DESCRIPTION
**Pull request by cartermckinnon**
**Issue #, if available:**

Closes #857

**Description of changes:**

This upgrades the 1.24 AMI's to use the 5.10 kernel line.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

I did `make 1.23`, confirmed it's still on 5.4; then did `make 1.24` and confirmed it's now on 5.10.

@camrakin TODO

### CODE REVIEW

File: scripts/install-worker.sh
Comment (cartermckinnon) [inline scripts/install-worker.sh:57]: I installed these in a new provisioner so they're available during `upgrade_kernel.sh`.

**Review by guessi — APPROVED** [review]

File: scripts/upgrade_kernel.sh
Comment (guessi) [inline scripts/upgrade_kernel.sh:15]: 🔽  would it be simpler like this? 

```
if [[ -z "$KERNEL_VERSION" ]]; then
  KERNEL_VERSION=5.4

  echo "kernel_version is unset. Setting to $KERNEL_VERSION"
  if vercmp "$KUBERNETES_VERSION" gteq "1.24.0"; then
    KERNEL_VERSION=5.10
  fi

  echo "kernel_version is unset. Setting to $KERNEL_VERSION based on Kubernetes version $KUBERNETES_VERSION."

fi
```

Comment (cartermckinnon) [reply scripts/upgrade_kernel.sh:15]: I don't think we need both the `echo`-s, but it does make sense to log the k8s version either way

**Review by guessi — APPROVED** [review]

**Comment by cartermckinnon** [comment]
@mhulscher I don't think the GPU operator specifically supports Amazon Linux at this point, though that support is being worked on as far as I know. My understanding is that the operator installs pre-compiled kernel modules, and I assume you'd run into issues if there was no such module compiled against the 5.10/5.15 headers. Just a guess; I'd probably open an issue with the operator maintainers.

**Comment by learhy** [comment]
@cartermckinnon any updates on this PR? Do you have an estimate as to when it will land in 1.24?

Thanks!

**Comment by RainofTerra** [comment]
We're currently building our own AMI that pulls in this change so we'd love to see it merged.

I'll probably file a separate issue for this, but it would be nice for 5.15 to be included as an option that we can pick at build time just so I don't have to maintain a (minor) fork to test later kernels on our nodes.

**Comment by cartermckinnon** [comment]
@RainofTerra yep, I plan to make that change once we get this merged 👍

**Comment by stacybird** [comment]
can't wait to see this merged.  :-)

**Review by camrakin — APPROVED** [review]

**Comment by bryantbiggs** [comment]
oh dang - is it happening?!

**Comment by learhy** [comment]
Hurrah!

**Comment by camrakin** [comment]
We’re ready to merge this! We compared the 5.4 and 5.10 kernels’:


1. Conformance, using Kubernete’s end-to-end integration test suite, the aws-k8s-tester, and long-running synthetic workloads.
    1. We observed no functional difference between the two kernels when it comes to Kubernetes.
2. Performance, using stress-ng and netperf. 
    1. We observed both marginal increases and decreases across the test cases; overall representing equivalent performance.

We’re excited for users to leverage the features of 5.10 in their clusters! We encourage users to monitor their workloads for differences when upgrading, and to open an issue here if something seems off.

**Comment by learhy** [comment]
Thanks, all!

**Comment by stevehipwell** [comment]
@cartermckinnon based on the release notes am I correct in thinking this didn't make it into v20230203?

**Comment by dims** [comment]
xref - https://github.com/aws/containers-roadmap/issues/1570

**Comment by cartermckinnon** [comment]
@stevehipwell Correct, this will land in the next release. Should be in the next week or so.

**Pull request merged**
